<?xml version="1.0" ?>
<launch>
  <arg name="robot_ip" />
  <arg name="load_gripper" default="true" />
  <arg name="arm_id" default="panda" />
  <arg name="gui" default="true" />

  <rosparam file="$(find franka_description_calibrated)/robots/$(arg arm_id)/joint_limits.yaml" command="load"/>

  <include file="$(find franka_human_friendly_controllers)/launch/franka_control.launch" >
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <rosparam command="load" file="$(find franka_human_friendly_controllers)/config/franka_human_friendly_controllers.yaml" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_variable_impedance_controller"/>
  <group if="$(arg gui)">
    <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_human_friendly_controllers)/launch/rviz/franka_description_with_marker.rviz"/>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />
  </group>
</launch>
